### PR TITLE
Issue #3262433 by tbsiqueira: Create a new CM+ "quick access dashboard" for better user experience

### PR DESCRIPTION
**Pull request by tbsiqueira**
## Problem
Currently there are a lot of menu options that are not easily accessible to CM+ users. Usually they need to "hunt" down links since they are scattered around Open Social main menu.

## Solution
Create a menu structure, at level one of Open Social main menu with all relevant menu links for CM+ users, similar to the already existing link at "/admin/config" ( img1 ).

Besides creating a central place for options, this allows extensions to easily add items to this menu.

## Issue tracker
https://www.drupal.org/project/social/issues/3262433

## How to test
- [ ] Log in as site manager and check if you have access to the new dashboard menu item;

## Screenshots
N/A

## Release notes
Content manager and Site managers now have access to a new dashboard, directly on Open Social Core

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2755-dbrc14s9snhrakuhsajkrkvrq540b6nb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620277d2046a4b8d29b21004

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2755-dbrc14s9snhrakuhsajkrkvrq540b6nb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620277d2046a4b8d29b21004

File: modules/social_features/social_core/images/icon-cog.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg height="24" viewBox="0 0 16 16" width="24" xmlns="http://www.w3.org/2000/svg">
+    <path d="m15.426 9.249c.045-.327.076-.658.076-.998 0-.36-.035-.71-.086-1.056l-2.275-.293c-.115-.426-.283-.827-.498-1.201l1.396-1.808c-.416-.551-.906-1.039-1.459-1.452l-1.807 1.391c-.373-.215-.774-.383-1.2-.499l-.292-2.252c-.338-.048-.677-.081-1.029-.081s-.694.033-1.032.082l-.291 2.251c-.426.116-.826.284-1.2.499l-1.805-1.391c-.552.413-1.044.901-1.459 1.452l1.395 1.808c-.215.374-.383.774-.499 1.2l-2.276.294c-.05.346-.085.696-.085 1.056 0 .34.031.671.077.998l2.285.295c.115.426.284.826.499 1.2l-1.417 1.836c.411.55.896 1.038 1.443 1.452l1.842-1.42c.374.215.774.383 1.2.498l.298 2.311c.337.047.677.08 1.025.08s.688-.033 1.021-.08l.299-2.311c.426-.115.826-.283 1.201-.498l1.842 1.42c.547-.414 1.031-.902 1.443-1.452l-1.416-1.837c.215-.373.383-.773.498-1.199zm-7.174 1.514c-1.406 0-2.543-1.137-2.543-2.541 0-1.402 1.137-2.541 2.543-2.541 1.402 0 2.541 1.138 2.541 2.541 0 1.404-1.139 2.541-2.541 2.541z"/>
+</svg>
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/images/icon-cog.svg:3]:
```suggestion
</svg>

```

File: modules/social_features/social_core/social_core.install [hunk 2]
```diff
@@ -1526,3 +1526,11 @@ function social_core_update_11005(): string {
   // Output logged messages to related channel of update execution.
   return $updateHelper->logger()->output();
 }
+
+/**
+ * Add dashboard permission to SM/CM.
+ */
+function social_core_update_11006(): void {
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/social_core.install:1533]: Shouldn't this permission be added on install instead of update (or maybe both?) ?

Comment (tbsiqueira) [reply modules/social_features/social_core/social_core.install:1535]: Good point, added on both ;)

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2755-dbrc14s9snhrakuhsajkrkvrq540b6nb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620277d2046a4b8d29b21004

File: modules/social_features/social_core/images/icon-cog.svg (hunk 1 above)
Comment (tbsiqueira) [reply modules/social_features/social_core/images/icon-cog.svg]: Done ;)

File: modules/social_features/social_core/social_core.module [hunk 3]
```diff
@@ -1121,3 +1121,23 @@ function social_core_preprocess_status_messages(&$variables) {
...
+/**
+ * Implements hook_menu_links_discovered_alter().
+ */
+function social_core_menu_links_discovered_alter(array &$links): void {
+
+  /** @var \Drupal\block_content\Entity\BlockContent $block */
+  $block = \Drupal::entityTypeManager()->getStorage('block_content')->loadByProperties(['type' => 'hero_call_to_action_block']);
+  $block = reset($block);
+
+  if ($block) {
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/social_core.module:1134]: I think we need to make this a little more robust:
If a user follows the link, and decides to delete the block for whatever reason, the link is still there (cached?) but also, what if multiple `hero_call_to_action_blocks`
get added? Some could be added on a landingpage or dashboard, and don't represent the one added on the front-page. 

Maybe we can specify the Properties a bit better for that scenario.

Screenshot 2022-02-10 at 12 35 36

File: modules/social_features/social_core/assets/css/admin-toolbar.scss [hunk 4]
```diff
@@ -24,6 +24,18 @@ body[class*='gin'] {
...
 			}
+
+      .menu-item__social_core-dashboard {
+        a.toolbar-icon-social-core-dashboard {
+          &:before {
+            width: 24px;
+            height: 24px;
+            -webkit-mask-image: none;
+            mask-image: none;
+            background: url('../../images/icon-cog.svg') no-repeat 0/cover;
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/assets/css/admin-toolbar.scss:35]: Screenshot 2022-02-10 at 12 37 45

We're better of with the open cog item, also make sure that once you're hovering over this it takes in to account the active theme color (happy to show you)

**Comment by tbsiqueira** [comment]
Closing this PR in favour of #2764

**Pull request closed without merging**